src/widgets/table/table.rs

```diff
@@ -1016,6 +1024,96 @@ mod tests {
         assert_eq!(table.widths, vec![Constraint::Percentage(100)], "vec ref");
     }
 
+    #[cfg(test)]
+    mod state {
```

Comment:
this (mostly) belongs in table_state.rs - the exceptions are how the rendering affects the state, which belongs here. (List hasn't been split into multiple files - yet)

Reply:
Have moved the first set of tests across - should the latter two stay here?

Reply:
The other two are fine. Thanks for fixing this up